autoload/ale/handlers.vim

```diff
@@ -116,6 +114,45 @@ function! ale#handlers#HandlePEP8Format(buffer, lines) abort
     return l:output
 endfunction
 
+
+function! ale#handlers#HandleFlake8Format(buffer, lines) abort
+    " Matches patterns line the following:
+    "
+    " stdin:6:6: E111 indentation is not a multiple of four
+    "
+    " Uses g:ale_python_flake8_error_codes to decide the severity of the
+    " message
+    let l:pattern = '^' . s:path_pattern . ':\(\d\+\):\?\(\d\+\)\?: \(\([[:alpha:]]\)[[:alnum:]]\+\) \(.*\)$'
+    let l:output = []
+
+    for l:line in a:lines
+        let l:match = matchlist(l:line, l:pattern)
+
+        if len(l:match) == 0
+            continue
+        endif
+
+        let l:code = l:match[3]
+        if (l:code ==# 'W291' || l:code ==# 'W293')
+                    \ && !g:ale_warn_about_trailing_whitespace
+            " Skip warnings for trailing whitespace if the option is off.
+            continue
+        endif
+
+        call add(l:output, {
+        \   'bufnr': a:buffer,
+        \   'lnum': l:match[1] + 0,
+        \   'vcol': 0,
+        \   'col': l:match[2] + 0,
+        \   'text': l:code . ': ' . l:match[5],
+        \   'type': index(g:ale_python_flake8_error_codes , l:code) == -1 ? 'W' : 'E',
```

Comment:
I have some suggestions here, after thinking about this a little more.

I reckon we should create a `Dictionary` instead of a `List`, and the values in the dictionaries should be the error types. That way, you could map some things to errors, other things to warnings, and then later some things to style warnings. (`'S'`) That will be useful for configuring extra `flake8` plugins.

I also think we should look at the start of the error code to distinguish between warnings or errors, so some things will be handled correctly by default. Your changes here could turn some errors into warnings. So errors beginning with `'W`' should be treated as warnings by default, and maybe treat everything else like an error by default.

We should probably provide no codes by default. Reading all of the `F` errors above, I would call them _errors_, not _warnings_. It's probably a matter of opinion, so it should be configurable. I recommend merging the default `Dictionary` with the user-configured `Dictionary`.

I'd write these changes into the `HandlePEP8Format` function above, change the global variable name to `g:ale_python_error_codes`, and document it in the Python documentation. Add some Vader tests for it. (Set some codes, call the function for handling lines of output, check the objects returned.)